src/ini.pest

```diff
@@ -1,17 +1,25 @@
-ini = { SOI ~ bom? ~ (pair | comment | NEWLINE)* ~ section* ~ EOI }
-bom = _{ "\u{feff}" }
-WHITESPACE = _{ "\t" }
+ini = { SOI ~ bom? ~ (blank | pair | comment)* ~ section* }
+
+// lines
+blank = _{ "" ~ NEWLINE }
+pair = { key ~ "=" ~ value ~ eol }
+comment = { indicator ~ value? ~ eol }
 
-// section
-section = { " "* ~ "[" ~ name ~ "]" ~ " "* ~ body }
-name = @{ char+ }
-char = { ASCII_ALPHANUMERIC | "." | "_" | "/" | " " | ";" | "#" | "\\" }
-body = { (NEWLINE ~ (pair | comment)?)* }
+// sections
+section = { header ~ eol ~ body }
+body = { (blank | pair | comment)* }
 
-// key-value pair
-pair = { key ~ "=" ~ value }
-key = @{ char{1,50} }
-value = @{ (char | "="){0,255} }
+// tokens
+// a name must end with a closing bracket followed by EOL
+header = @{ "[" ~ (!("]" ~ (char | eol)) ~ char)+ ~ "]" }
```

Comment:
I had to include the brackets in this header token in order to preserve the spaces inbetween.